app/tests/test_process_observations.py

```diff
@@ -229,3 +229,123 @@ async def test_process_observation_cameratrap_and_route_to_gcp_pubsub_dispatcher
     # Check that the right methods, to publish to PuSub, were called
     assert mock_pubsub_client.PublisherClient.called
     assert mock_pubsub_client.PublisherClient.return_value.publish.called
+
+
+@pytest.mark.asyncio
+async def test_retry_process_event_on_smart_error_and_send_to_dlq(
+    mocker,
+    mock_cache,
+    mock_gundi_client,
+    mock_smart_async_client_class_with_server_error,
+    mock_send_observation_to_dead_letter_topic,
+    raw_observation_geoevent_with_valid_uuid,
+    raw_observation_geoevent_attributes,
+    smart_outbound_configuration_gcp_pubsub,
+):
+    # Override gundi client mock to return a smart destination
+    mock_gundi_client.get_outbound_integration.return_value = async_return(
+        smart_outbound_configuration_gcp_pubsub
+    )
+    mock_gundi_client.get_outbound_integration_list.return_value = async_return(
+        [smart_outbound_configuration_gcp_pubsub]
+    )
+    # Mock external dependencies
+    mocker.patch("app.core.gundi._cache_db", mock_cache)
+    mocker.patch("app.core.gundi._portal", mock_gundi_client)
+    mocker.patch(
+        "app.services.transformers.AsyncSmartClient",
+        mock_smart_async_client_class_with_server_error,
+    )
+    mocker.patch(
+        "app.services.process_messages.send_observation_to_dead_letter_topic",
+        mock_send_observation_to_dead_letter_topic
+    )
+    await process_observation(
+        raw_observation_geoevent_with_valid_uuid, raw_observation_geoevent_attributes
+    )
+
+    # Check that is retried and then sent to the dead letter topic
+    assert mock_smart_async_client_class_with_server_error.return_value.get_incident.call_count == 3
+    assert mock_send_observation_to_dead_letter_topic.called
+
+
+@pytest.mark.asyncio
+async def test_process_event_doesnt_call_smart_with_invalid_uuid(
```

Comment:
@marianobrc I wasn't quite sure where the mock failure should come from in this test case. Re `_with_invalid_uuid` were you trying to mock a bad `event.id` or a bad CA UUID?

Reply:
I was trying to mock a bad `event.id`, using the mock `raw_observation_geoevent` where it's set to None, as it's the common case for case any new events pushed through the sensors api.

Reply:
I found out why that test was failing. The mock didn't have an event type matching a smart category, and also there was a patch missing to mock the PubSub Publisher. All the tests are passing now.